lms/djangoapps/courseware/masquerade.py

```diff
@@ -128,17 +128,19 @@ def get(self, request, course_key_string):
                     'name': 'Staff',
                     'role': 'staff',
                 },
-                {
-                    'name': 'Learner',
-                    'role': 'student',
-                },
-                {
-                    'name': 'Specific Student...',
-                    'role': 'student',
-                    'user_name': course.user_name or '',
-                },
             ],
         }
+        if len(partitions) == 0:
+            data['available'].append({
+                'name': 'Learner',
+                'role': 'student',
+            })
+
+        data['available'].append({
+            'name': 'Specific Student...',
```

Comment:
Interesting that we don't translate these. Course teams see these - and they might reasonably not speak English in some open edx intalls. I know that's a bit of a blurry line - we generally assume installation owners speak English because we don't bother translating all our logging messages. But this feels a bit more on the user side.

Not something for you to fix here in this PR, but just noting.